azure-ipam/ipam.go

```diff
@@ -72,10 +74,29 @@ func (p *IPAMPlugin) CmdAdd(args *cniSkel.CmdArgs) error {
 	p.logger.Debug("Making request to CNS")
 	// if this fails, the caller plugin should execute again with cmdDel before returning error.
 	// https://www.cni.dev/docs/spec/#delegated-plugin-execution-procedure
-	resp, err := p.cnsClient.RequestIPAddress(context.TODO(), req)
+	resp, err := p.cnsClient.RequestIPs(context.TODO(), req)
 	if err != nil {
-		p.logger.Error("Failed to request IP address from CNS", zap.Error(err), zap.Any("request", req))
-		return cniTypes.NewError(ErrRequestIPConfigFromCNS, err.Error(), "failed to request IP address from CNS")
+		// if we fail a request with a 404 error try using the old API
+		p.logger.Error("Failed to request IPs using RequestIPs from CNS, going to try RequestIPAddress", zap.Error(err), zap.Any("request", req))
```

Comment:
I think this log line belongs inside the `if` clause that starts on the next line

Reply:
Moved it inside so that it now only prints when we get 404